src/Microsoft.AspNetCore.Mvc.Testing/Handlers/RedirectHandler.cs

```diff
@@ -0,0 +1,139 @@
+            CopyContentHeaders(originalRequestContent, request.Content, contentCopy);
+
+            return contentCopy;
+        }
+
+        private static void CopyContentHeaders(
+            HttpContent originalRequestContent,
+            HttpContent newRequestContent,
+            HttpContent contentCopy)
+        {
+            foreach (var header in originalRequestContent.Headers)
+            {
+                contentCopy.Headers.Add(header.Key, header.Value);
+                newRequestContent.Headers.Add(header.Key, header.Value);
+            }
+        }
+
+        private static async Task<(Stream originalBody, Stream copy)> CopyBody(HttpRequestMessage request)
+        {
+            var originalBody = await request.Content.ReadAsStreamAsync();
+            var bodyCopy = new MemoryStream();
+            await originalBody.CopyToAsync(bodyCopy);
+            bodyCopy.Seek(0, SeekOrigin.Begin);
+            if (originalBody.CanSeek)
+            {
+                originalBody.Seek(0, SeekOrigin.Begin);
+            }
+            else
+            {
+                originalBody = new MemoryStream();
+                await bodyCopy.CopyToAsync(originalBody);
+                originalBody.Seek(0, SeekOrigin.Begin);
+                bodyCopy.Seek(0, SeekOrigin.Begin);
+            }
+
+            return (originalBody, bodyCopy);
+        }
+
+        private static HttpRequestMessage GetRedirectRequest(
+            HttpResponseMessage response,
+            HttpContent originalContent)
+        {
+            var location = response.Headers.Location;
+            if (!location.IsAbsoluteUri)
+            {
+                location = new Uri(
+                    new Uri(response.RequestMessage.RequestUri.GetLeftPart(UriPartial.Authority)),
+                    location);
+            }
+
+            var redirect = !ShouldKeepVerb(response) ?
+                new HttpRequestMessage(HttpMethod.Get, location) :
+                new HttpRequestMessage(response.RequestMessage.Method, location)
+                {
+                    Content = originalContent
+                };
+
+            foreach (var header in response.RequestMessage.Headers)
+            {
+                redirect.Headers.Add(header.Key, header.Value);
+            }
+
+            foreach (var property in response.RequestMessage.Properties)
+            {
+                redirect.Properties.Add(property.Key, property.Value);
+            }
+
+            return redirect;
+        }
+
+        private static bool ShouldKeepVerb(HttpResponseMessage response) =>
+            response.StatusCode == HttpStatusCode.RedirectKeepVerb ||
+                (int)response.StatusCode == 308;
+
+        private bool IsRedirect(HttpResponseMessage response) => 
+            response.StatusCode == HttpStatusCode.MovedPermanently ||
+                response.StatusCode == HttpStatusCode.Redirect ||
+                response.StatusCode == HttpStatusCode.TemporaryRedirect ||
```

Comment:
Maybe use `HttpStatusCode.RedirectKeepVerb` here for consistency? (Not sure why there are two `307` values in the enum...)

Reply:
This enum is outdated and evil